### DISCUSSION TITLE
TTL for logs , traces and metrics

### DISCUSSION THREAD

**Discussion by basanth-ivoyant** [SigNoz/signoz#6436] [General]
Hi,

We have single Signoz setup for all environments, as we can't do true multitenancy, we are using tags to filter the records by environment. We might at some point of time have different setup for different environments.

Is there a way we can set a higher TTL at 90 days, so for all entries the default TTL is 90 days. Is it possible to search by tag and either reset the ttl for those entries or delete them using a cron  job?

regards,
Basanth

**Comment by srikanthccv** [comment]
> Is it possible to search by tag and either reset the ttl for those entries or delete them using a cron job?

Today, there is no such capability in SigNoz. Technically it can be achieved by doing some CH mutations. This will effectively be a delete operation because you must conditionally delete data. But CH does the worst job with mutation so I would encourage not to run any mutation.
